From 3b20f3ec70ba9d7f6c2092faad93d1dbe322ca16 Mon Sep 17 00:00:00 2001
From: Nishad Saraf <nishad.saraf@xilinx.com>
Date: Wed, 9 Sep 2020 23:46:19 -0700
Subject: [PATCH 1580/1851] dt-bindings: soc: xilinx: ai-engine: Add interrupt
 and power-domain properties for AIE device

commit 376740c2e47988382dabb33c0a06a1dbbdf49b25 from
https://github.com/Xilinx/linux-xlnx.git

Add interrupt and power-domain properties to support interrupt
handling and platform management services.

Signed-off-by: Nishad Saraf <nishad.saraf@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/soc/xilinx/xlnx,ai-engine.yaml    | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml b/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
index 3292ac2ecd2a..25bbede01608 100644
--- a/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
+++ b/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
@@ -39,11 +39,22 @@ properties:
       size of cell to describe AI engine range of tiles size.
       As the AI engine tiles are 2D array, the size cell is 2.
 
+  interrupts:
+    maxItems: 3
+
+  interrupt-names:
+    description: |
+      Should be "interrupt1", "interrupt2" or "interrupt3".
+
 required:
   - compatible
   - reg
   - '#address-cells'
   - '#size-cells'
+  - power-domains
+  - interrupt-parent
+  - interrupts
+  - interrupt-names
 
 patternProperties:
   "^partition[0-9]@[0-9]+$":
@@ -82,11 +93,18 @@ patternProperties:
 
 examples:
   - |
+    #include <dt-bindings/power/xlnx-versal-power.h>
     ai_engine: ai-engine@20000000000 {
       compatible = "xlnx,ai-engine-v1.0";
       reg = <0x200 0x0 0x1 0x0>;
       #address-cells = <2>;
       #size-cells = <2>;
+      power-domains = <&versal_firmware PM_DEV_AI>;
+      interrupt-parent = <&gic>;
+      interrupts = <0x0 0x94 0x4>,
+                   <0x0 0x95 0x4>,
+                   <0x0 0x96 0x4>;
+      interrupt-names = "interrupt1", "interrupt2", "interrupt3";
 
       aie_partition0: aie-partition@0 {
         /* 50 columns and 8 core tile rows + 1 SHIM row */
-- 
2.31.1

